### PR TITLE
[4.4] Add Joomla 5 compat plugin to 4.4.0 to fix class "JPlugin" not found error on update to 5

### PR DESCRIPTION
**Pull request by richard67**
Pull Request for Issue # .

### Summary of Changes

This pull request (PR) adds the compatibility plugin from 5.0 to 4.4 so that it's available when updating to 5.0.

The plugin as such is more or less empty so it doesn't do anything on 4.4.

It needs to be there when updating to 5.0 before the database updates run, otherwise you get an exception when having a system plugin installed and enabled which descends from JPlugin.

#### Test 1: Reproduce the issue

1. On a clean 4.4-dev or latest 4.4 nightly build or 4.4-alpha4 installation, install the following plugin and enable it after installation: https://test5.richard-fath.de/ttest2.zip

2. Update to the latest 5.0 nightly build or an update package build from the current 5.0-dev branch.

Result: See section "Actual result BEFORE applying this Pull Request" below.

#### Test 2: Test the fix from this PR with a new 4.4 installation with the fix

1. On a clean 4.4-dev or latest 4.4 nightly build or 4.4-alpha4, apply the patch of this PR.

2. Make a new installation.

3. Install the following plugin and enable it after installation: https://test5.richard-fath.de/ttest2.zip

4. Update to the patched package or custom update URL created by Drone for PR #41739 .

Result: See section "Expected result AFTER applying this Pull Request" below.

#### Test 3: Test the fix from this PR with a 4.4 previously updated to the 4.4 with the fix

1. On a clean 4.4-dev or latest 4.4 nightly build or 4.4-alpha4 installation without the patch of this PR applied, update to the patched package or custom update URL created by Drone for this PR here.
 
2. Install the following plugin and enable it after installation: https://test5.richard-fath.de/ttest2.zip

3. Update to the patched package or custom update URL created by drone for PR #41739 .

Result: See section "Expected result AFTER applying this Pull Request" below.

### Actual result BEFORE applying this Pull Request

Depending on your global configuration settings for "Error Reporting" and "Debug System", you get the following results:
- When "Error Reporting" is set to "Default" or "None" and "Debug System" is switched off, you just get an internal server error 500 without any information about the reason. The update log "administrator/logs/joomla_update.php" ends after the update package has been unpacked, i.e. the database updates and other steps have not run.
- When "Error Reporting" is set to "Maximum" and "Debug System" is switched on, you will see 2 errors, 1. an SQL error caused by the SQL updates not having run, and 2. an error `ERROR Class "JPlugin" not found`. The latter error is the reason why the finalization step of the update is not started and the database updates have not run.

### Expected result AFTER applying this Pull Request

The update succeeds.

### Link to documentations
Please select:
- [X] No documentation changes for docs.joomla.org needed

- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by laoneo** [comment]
Is this needed only for the class JPlugin or all J* classes?

**Comment by Fedik** [comment]
For all J* classes. Any system plugin still may use one of J* class in constructor or in other system events.

File: plugins/behaviour/compat/compat.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<extension type="plugin" group="behaviour" method="upgrade">
+	<name>plg_behaviour_compat</name>
+	<author>Joomla! Project</author>
+	<creationDate>2023-03</creationDate>
```
Comment (brianteeman) [inline plugins/behaviour/compat/compat.xml]:
```suggestion
	<creationDate>2023-09</creationDate>
```

Comment (richard67) [reply plugins/behaviour/compat/compat.xml]: In general I agree, but then we will have the plugin in J5 with an older creationDate than in J4. Maybe that could confuse people who like to read XML files?

Comment (richard67) [reply plugins/behaviour/compat/compat.xml]: And when we merge this change up, the plugin will suddently become 6 months younger in J5.

Comment (richard67) [reply plugins/behaviour/compat/compat.xml]: @laoneo What's your advice?

Comment (richard67) [reply plugins/behaviour/compat/compat.xml]: Hmm, I think we should accept this suggestion and keep the date in 5.0-dev when merging up, so you can see it was back integrated from 5.0-dev into 4.0-dev.

**Comment by richard67** [comment]
To be honest, I am not very happy with my solution, but it seems there is no other way to fix the issue, or any other way would be worse.

File: plugins/behaviour/compat/src/Extension/Compat.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
+<?php
```
Comment (richard67) [inline plugins/behaviour/compat/src/Extension/Compat.php:1]: Shouldn't this file be called "Compat6.php"?

**Pull request merged**